lib/autoupdate/rollout/strategy_timebased.go

```diff
@@ -51,7 +51,13 @@ func newTimeBasedStrategy(log *slog.Logger) (rolloutStrategy, error) {
 	}, nil
 }
 
-func (h *timeBasedStrategy) progressRollout(ctx context.Context, status *autoupdate.AutoUpdateAgentRolloutStatus, now time.Time) error {
+func (h *timeBasedStrategy) progressRollout(ctx context.Context, spec *autoupdate.AutoUpdateAgentRolloutSpec, status *autoupdate.AutoUpdateAgentRolloutStatus, now time.Time) error {
+	windowDuration := spec.MaintenanceWindowDuration.AsDuration()
```

Comment:
Note that a nil spec would panic this line. (Not sure if intentional or not, so adding the comment to clarify.)

Reply:
Fixed, now we handle the case gracefully.